### PR TITLE
Add persistent cookie handling for remembered device sign in

### PR DESCRIPTION
**Pull request by pauldoomgov**
* sign_in helper records users as they visit, then starts replaying after
  threhold is met, attempting to hit remembered sign in target.
* sign_in helper uses per-user cookie cache to restore
  "user_opted_remember_device_preference" and "remember_device"
  cookies for a given user.

Able to push over 90% remembered device mix.

### CODE REVIEW

**Comment by pauldoomgov** [comment]
Apologies for the amount of reformatting in this PR.  I do recommend use of the `black` formatter for consistency.

File: load_testing/common_flows/flow_helper.py [hunk 1]
```diff
@@ -167,23 +178,61 @@ def random_cred(num_users):
...
+        "email": f"testuser{user_num}@example.com",
         "password": "salty pickles",
     }
 
     return credential
 
+
+def use_previous_visitor(visited_count, visited_min, remembered_target):
+    """
+    Helper to decide if a specific sign in should use a previously used used.
```
Comment (ctro) [inline load_testing/common_flows/flow_helper.py:211]: Previously used _user_ I think, everything else looks great so far!

File: load_testing/common_flows/flow_helper.py [hunk 2]
```diff
@@ -222,3 +273,44 @@ def load_fixture(filename, path="./load_testing"):
...
+        savelist = DEFAULT_COOKIE_SAVELIST
+
+    # Pulling directly from internal data structure as there is
+    # no get_cookies method.
+    dcookies = cookies._cookies.get(domain, {}).get('/', None)
+
+    if dcookies is None:
+        return []
+
+    return [c for c in [dcookies.get(si) for si in savelist] if c is not None]
```
Comment (ctro) [inline load_testing/common_flows/flow_helper.py:301]: I sang this line to the tune of "C is for 🍪"

File: tests/test_flow_helpers.py [hunk 3]
```diff
@@ -22,56 +26,109 @@
...
 
+
 def test_authentication_token():
     resp = test_helper.mock_response("doc_auth_verify.html")
 
-    assert authenticity_token(resp) == "WPhfbuwqPzfbpB2+aTHWR93t0/7O88iK5nYdL/RaZoLEPH63Cjf4yKAkHw6CUDyaXw6O5oi4Nc2NHzC6stEdwA=="
-    assert authenticity_token(resp, 0)  == "WPhfbuwqPzfbpB2+aTHWR93t0/7O88iK5nYdL/RaZoLEPH63Cjf4yKAkHw6CUDyaXw6O5oi4Nc2NHzC6stEdwA=="
-    assert authenticity_token(resp, 1) == "I7WOA3x24rsZVj56R9QtCNVNlXapxqo2A9MOkU2sHPIsAi99KMzwSzD3Y89H710hluHXCoKOYt8VkT77f9U/Kg=="
-    assert authenticity_token(resp, 2) == "679gwHHowpDvKlzyBL4Cw2MYZC1NYLqWaAEz+Nze6ZJZELBdu1t7BTlGmVkvqfBh713/xc0oCkbndTMoOlpLRg=="
+    assert (
```
Comment (ctro) [inline tests/test_flow_helpers.py:111]: Thanks for all the formatting all over the place.

File: load_testing/common_flows/flow_helper.py (hunk 2 above)
Comment (pauldoomgov) [reply load_testing/common_flows/flow_helper.py:301]: Good enough for me.

File: load_testing/common_flows/flow_helper.py (hunk 1 above)
Comment (pauldoomgov) [reply load_testing/common_flows/flow_helper.py:211]: Good eyes!

**Pull request merged**